src/frontend/query_commands.ml

```diff
@@ -359,6 +359,43 @@ let dispatch pipeline (type a) : a Query_protocol.t -> a =
     in
     List.map ~f:Mbrowse.node_loc path
 
+  | Locate_type pos ->
+    let typer = Mpipeline.typer_result pipeline in
+    let structures = Mbrowse.of_typedtree (Mtyper.get_typedtree typer) in
+    let pos = Mpipeline.get_lexing_pos pipeline pos in
+    let path = Mbrowse.enclosing pos [structures] in
+    let path =
+      Std.List.filter_map path ~f:(fun (env, node) ->
+        Locate.log ~title:"debug" "inspecting node: %s" (Browse_raw.string_of_node node);
+        match node with
+        | Browse_raw.Expression {exp_type = ty; _}
+        | Pattern {pat_type = ty; _}
+        | Core_type {ctyp_type = ty; _}
+        | Value_description { val_desc = { ctyp_type = ty; _ }; _ } ->
+          begin match (Ctype.repr ty).desc with
+            | Tconstr (path, _, _) -> Some (env, path)
+            | _ -> None
+          end
+        | _ -> None
+      )
+    in
+    Std.List.filter_map path ~f:(fun (_env, path) ->
+      Locate.log ~title:"debug" "found type: %s" (Path.name path);
+      let local_defs = Mtyper.get_typedtree typer in
+      match
+        Locate.from_path
+          ~config:(Mpipeline.final_config pipeline)
+          ~local_defs ~pos ~namespace:`Type `MLI
+          path
+      with
+      | exception Env.Error _ -> None
+      | `Found (path, lex_position) -> Some (path, lex_position)
+      | `File_not_found _
+      | `Invalid_context
+      | `Missing_labels_namespace
+      | `Not_found _
+      | `Not_in_env _ -> None)
```

Comment:
I assume you're swallowing the error because of the `filter_map` call, but that means that when you don't find any result, the user won't know why.

Reply:
Fixed